src/main/java/seedu/hms/model/booking/Booking.java

```diff
@@ -69,10 +69,36 @@ public int numOfCustomers() {
 
     @Override
     public String toString() {
+        String otherUsersToString = "";
+        if (otherUsers.isPresent()) {
+            for (Customer u : otherUsers.get()) {
+                otherUsersToString += u.getName();
+            }
+        }
         return "Booking for "
             + service.getName()
             + " from " + timing
             + " by " + payer.getName()
+            + " with " + otherUsersToString
             + ". Comment - " + comment.orElse("N/A");
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Booking)) {
+            return false;
+        }
+
+        Booking ob = (Booking) other;
+        System.out.println(getOtherUsers());
+        System.out.println(ob.getOtherUsers());
+        return ob.getService().equals(getService())
+                && ob.getTiming().equals(getTiming())
+                && ob.getPayer().equals(getPayer())
+                && ob.getOtherUsers().equals(getOtherUsers());
```

Comment:
Does this check the contents of the optional lists?

Reply:
Yes, it's automatically handled